docs/developers.md

```diff
@@ -418,6 +418,89 @@ etherpad_db_host: "{{ lookup('env','ETHERPAD_DB_HOST') | default('mariadb', true
 state: present
 ```
 
+### Alternative to using `apb push`
+When developing APBs, there are a couple of factors which could prevent the developer from using the full development lifecycle that the APB tooling offers. Primarily these factors are:
+* Developing against an OpenShift/Kubernetes cluster that exists on a remote host
+* Developing APBs on a machine that doesn't have access to the Docker daemon
+
+If a developer meets any of these criteria, then we suggest the following workflow to publish images to the internal OCP registry so that the Automation Broker can bootstrap the image. This section will show you how to do these steps with the APB tooling and without.
+
+* Step 1: Ensure the base64 encoded spec is a label on the Dockerfile
```

Comment:
minor suggestion, but I think of this as being "in" the Dockerfile, not "on". It's not a property of the file, but part of its contents.